src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs

```diff
@@ -133,7 +133,7 @@ public override bool Execute()
                     () => RestoreSourcesOverride?.Select(MSBuildRestoreUtility.FixSourcePath).Select(e => GetGlobalAbsolutePath(e)).ToArray(),
                     () => MSBuildRestoreUtility.ContainsClearKeyword(RestoreSources) ? Array.Empty<string>() : null,
                     () => RestoreSources?.Select(MSBuildRestoreUtility.FixSourcePath).Select(e => UriUtility.GetAbsolutePathFromFile(ProjectUniqueName, e)).ToArray(),
-                    () => (new PackageSourceProvider(settings)).LoadPackageSources().Where(e => e.IsEnabled).Select(e => e.Source).ToArray());
+                    () => (new PackageSourceProvider(settings, enablePackageSourcesChangedEvent: false)).LoadPackageSources().Where(e => e.IsEnabled).Select(e => e.Source).ToArray());
```

Comment:
In many places in this PR there are instantiations of `PackageSourceProvider` (with the new argument) just to call `LoadPackageSources()`.

Consider having a static (internal?) method (`PackageSourceProvider.LoadPackageSources(ISettings)`) which never subscribes events and eliminates the allocation for the throwaway `PackageSourceProvider` instance.  Within `PackageSourceProvider` you can share implementation with the constructor or leave it to https://github.com/NuGet/Home/issues/8479 fix.

Reply:
There's something that makes me a bit uncomfortable about the default being the more resource-intensive one and having to pepper this extra parameter all throughout the code (and remember to do so), esp. since there doesn't seem any disposer that will remove the event right now.

I think this would very well be fixed by @dtivel's suggestion above, too. Are all the relevant cases where the change event is disabled the ones where `#LoadPackageSources()` is called?